### ISSUE TITLE
[Bug]: Firefox Flatpak package can't use symlinks in Downloads folder alongside some addons (Ex: Save In)

### ISSUE THREAD

**Issue by LukasThyWalls** [flatpak/flatpak#4994]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

1.12.7

### What Linux distribution are you using?

Ubuntu

### Linux distribution version

KUbuntu 22.04 (5.15.0-41-generic)

### What architecture are you using?

x86_64

### How to reproduce

Because Firefox decided at some point to not allow set path outside the Downloads folder for addons ( https://bugzilla.mozilla.org/show_bug.cgi?id=1342563 ) so some addons related to Download managing set a workaround meanwhile there isn't another option, what is setting symlinks in the Download folder to wherever you want to save a files with the addons.

1)

As an example, i use Save In https://addons.mozilla.org/en-US/firefox/addon/save-in/ what adds the option to save everything in a customizable menu in the context menu automatically without prompt.

2)

In the Downloads Folder i have this:
* _Firefox_WebEx_, a (hidden) folder
* _Firefox_WebEx/USBDisk_, a symlink to /media/"myuser"/USBDisk, a mounting point to a USB Disk that i plug in when i need it.
* _Firefox_WebEx/Desktop_, a symlink to /home/"myuser"/Desktop

All the symlinks work with the file explorer (Dolphin).

3)

In the settings inside Save In, i have this, allowing to save everywhere to test:

```
Firefox_WebEx/USBDisk/Downloads // symlink, Downloads is an existing folder inside the USB Disk
Firefox_WebEx/Desktop // symlink
. // (alias: Downloads)
---
Firefox_WebEx // parent workaround folder
Firefox_WebEx/Test // regular folder
```

### Expected Behavior

With the Firefox Snap Package, and with Firefox on Windows, the setup with the symlinks works.
Choosing in the Save In' context menu, you can save an image, for example, in every folder and subfolders configured without errors, including going throw the symlinks to the Desktop and the USB Disk. They appear in the Firefox Download Manager as successful downloads.
Note: When the USB Disk is not mounted/plugged in, trying to save there becomes an failed download and bring the save dialog to choose where to save it (that's Save In addon behaviour).

### Actual Behavior

Meanwhile with the Firefox Flatpak package, but regular folders and subfolders works, but the symlinks doesn't work.
Choosing the paths without symlinks, they work fine, they save an image, for example, in the selected folder.
Choosing the paths with symlinks, however, doesn't work.
* The USB Disk one, there is no error or notification of failed download, either appear in the Firefox Download Manager.
* The Desktop one, Firefox/Save In throw a FILE_FAILED and a Save dialog to choose path to save the file.
Curiously, with the save dialog i can go to the symlinks in the Firefox_WebEx folder, go throw any of them, and save the file successfully. But doing it with the addon is no allowed.

### Additional Information

* I reported to the Save In issue tracker here https://github.com/gyng/save-in/issues/185, to gather information because i don't think is directly an issue of Save In, instead as how Flatpak isolates the Firefox package and how that affect the webextensions engine.
* I was using the Firefox Snap package for months with this configuration, working without issue, but when i changed it to the flatpak package, i detected the change of behaviour. First, I imported my old Firefox profiles to the Flatpak version, and saw the issue, but later i reproduced it in a clean profile to avoid if something else was interfering.
* In the Save In Issue, they told me to test if with _bindfs_, mounting a folder in other folder, the issue is gone or not. It works, also with the Flatpak version, but you need to install extra packages, and introduces an issue: You can't use it with the USB Disk because it needs to be mounted to create the mounting point, and the USB Disk refuse to be unmount/plugged out securely because the mounting point is still active, so you can't use seamlessly like with the symlinks, as you need to mount and umount the folder every time you plug the USB Disk.

**Comment by JakobDev** [comment]
Maybe your Firefox Flatpak has just access to the Downloads folder

**Comment by LukasThyWalls** [comment]
> Maybe your Firefox Flatpak has just access to the Downloads folder

Hello! I didn't remark that in the behaviour (i edited it now), but I also created regular folders and subfolders in the Downloads folder and the addon can save in those folders normally, the only thing that doesn't do is follow the symlinks.

**Comment by LukasThyWalls** [comment]
Thanks to point me to the right direction.

As you said, it's because how flatpak isolates the filesystem, and you can't cheat it with symlinks. With Flatseal, I set access for Firefox to all system (filesystem=host) and with that it works! (It also began to use the profiles in ~/.mozilla instead the flatpak ones, making a little mess). But because i don't think is needed to Firefox to see all the filesystem, i configured with Flatseal access only to _/media/<myuser>/_ (to access all USB, although i can be more specific) and _/home/<myuser>/Desktop_, and with that it follows the symlinks and Save In works as intended.

I think this could set as fixed/closed with this, but i don't know if this need some kind of clarification, because the Firefox Flatpak package needs extra settings to do the same things as other packages, and there isn't any error or hint of what is the issue or what to do to fix it. Or maybe adding an option to follow symlinks by default in this package, although i don't see it appropriate for the concept of what a Flatpak is. However, is a bit weird because with this we need to "trick" Firefox to reach outside the Downloads folder, and report the trick to the trick to Flatpak to make it work. 

The root of this is in the decision of Firefox/Webextensions to not allow the addons to reach anything outside the Downloads folder and still not add some way to add permissions to addons to specific folders (something discussed here https://bugzilla.mozilla.org/show_bug.cgi?id=1342563 ) and you need to do workarounds to make this work. With a proper permissions by addon, maybe you can spread the configuration and give to the flatpak configuration the path the app needs to access and nothing more.
No criticisim to Firefox devs decisions, only thinking out loud about where this issue comes.

Thanks!

**Comment by chrisawi** [comment]
Yeah, the Flatpak sandbox doesn't allow apps to use symlinks to access directories outside the sandbox.
